process/sync/baseSync.go

```diff
@@ -498,20 +509,17 @@ func (boot *baseBootstrap) removeHeaderFromPools(header data.HeaderHandler) []by
 		return nil
 	}
 
-	//TODO: boot.headers.Remove(hash) should not be called, just to have a restore point if it is needed later
 	return hash
 }
 
-func (boot *baseBootstrap) cleanCachesAndStorageOnRollback(
-	header data.HeaderHandler,
-	headerStore storage.Storer,
-	headerNonceHashStore storage.Storer) {
-
+func (boot *baseBootstrap) cleanCachesAndStorageOnRollback(header data.HeaderHandler) {
 	hash := boot.removeHeaderFromPools(header)
 	boot.forkDetector.RemoveHeaders(header.GetNonce(), hash)
-	_ = headerStore.Remove(hash)
+	//TODO: Refactor the deletion from the headerStore as that will be an exceptional case in which,
+	//in case of "bad" rollbacks and datapool accidental eviction, the shard will halt.
+	_ = boot.headerStore.Remove(hash)
 	nonceToByteSlice := boot.uint64Converter.ToByteSlice(header.GetNonce())
-	_ = headerNonceHashStore.Remove(nonceToByteSlice)
+	_ = boot.headerNonceHashStore.Remove(nonceToByteSlice)
```

Comment:
do store remove first, than you can see if there are erros. In case of errors you do not remove from pool

Reply:
I think it does't matter the order, and if I can or not to delete it from storage, as anyway this block would be rollback and I want to clean its appearance from every where, even if in some storages or pools it does't exist anymore or there are some errors when these cleaning actions would be done.